### PR TITLE
feat(API-1953): Add guided tutorial "Guided tutorial: How to collect product variations"

### PR DESCRIPTION
**Pull request by samokiss**
https://akeneo.atlassian.net/browse/API-1903

### CODE REVIEW

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+```php [activate:PHP]
+
+function getProductModels(): array
+{
+    $client = getApiClient();
+
+    $maxProductsPerPage = 100;
+    $maxFamiliesPerQuery = 3;
+    $locales = 'fr_FR';
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:132]: locales come from your storage (you stored them in https://api.akeneo.com/tutorials/how-to-retrieve-pim-structure.html)

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    $scope = 'ecommerce';
+
+    $familyCodeChunks = array_chunk(getFamilyCodes(), $maxFamiliesPerQuery);
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:135]: `getFamilyCodes()`, what is it? I know it is the method to get family codes from your storage, but the developer who reads this GT could feel puzzled.
I suggest doing the same thing as in https://api.akeneo.com/tutorials/how-to-get-families-and-attributes.html and precise what is this function by adding a comment

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+#### 1. Collect product models
+##### 1.1 You are following the App workflow?
+
+In the guided tutorial How to get families and attributes, we have stored a family_code_list. It’s time to use it!
+
+```php [activate:PHP]
+
+function getProductModels(): array
+{
+    $client = getApiClient();
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:128]: I guess this is the initialization part? If yes, you should change your step 0 and define` getApiClient()`

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+        $response = $client->get(sprintf($apiUrl, $locales, $scope, json_encode($familyCodes)));
+        $data = json_decode($response->getBody()->getContents(), true);
+        $productModels[] = $data['_embedded']['items'];
+    }
+
+    $productModels = array_merge(...$productModels);
+
+    saveProductModels($productModels);
+
+    return $productModels;
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:156]: Why do you return `$productModels`?

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+        $response = $client->get($data['_links']['next']['href']);
+        $data = json_decode($response->getBody()->getContents(), true);
+        $productModels[] = $data['_embedded']['items'];
+    }
+
+    $productModels = array_merge(...$productModels);
+
+    saveProductModels($productModels);
+
+    return $productModels;
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:195]: Why do you return `$productModels`?

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+Simply get the attribute type by requesting the API
+
+```php [activate:PHP]
+
+function getProductModels(): array
+{
+    $client = getApiClient();
+
+    $maxProductsPerPage = 100;
+    $locales = 'fr_FR';
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:171]: No filter on locales

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+    $scope = 'ecommerce';
+
+    $apiUrl = '/api/rest/v1/product-models?'
+        . 'locales=%s'
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:175]: No locales --> https://www.notion.so/akeneo/Guided-tutorial-How-to-collect-product-variations-183f7b14a75147788336d48fbcaf7c7f

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+```
+
+##### 1.2 You are not following the App workflow?
+Simply get the attribute type by requesting the API
+
+```php [activate:PHP]
+
+function getProductModels(): array
+{
+    $client = getApiClient();
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:168]: I guess this is the initialization part? If yes, you should change your step 0 and define `getApiClient()`

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+Good news: you already store the family variant in the guided tutorial **“How to get families and attributes”**. Go ahead!
+
+###### 2.2.2 You are not following the App workflow?
+Query the API.
+
+```php [activate:PHP]
+
+function getFamilyVariants(): array
+{
+    $client = getApiClient();
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:216]: I guess this is the initialization part? If yes, you should change your step 0 and define `getApiClient()`

**Review by tomglvng — CHANGES_REQUESTED** [review]

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+```
+
+#### 2. Process product model
+##### 2.1. Parse and store the product model
+Parse and store a product or a product model is definitely the same thing. Please have a how to our guided tutorial **“How to get product information”**.
+
+##### 2.2. Parse and store the product model
+###### 2.2.1 You are following the App workflow?
+
+Good news: you already store the family variant in the guided tutorial **“How to get families and attributes”**. Go ahead!
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:207]: Missing link

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+    saveProductModels($productModels);
+
+    return $productModels;
+}
+
+```
+
+#### 2. Process product model
+##### 2.1. Parse and store the product model
+Parse and store a product or a product model is definitely the same thing. Please have a how to our guided tutorial **“How to get product information”**.
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:202]: missing link

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 12]
```diff
@@ -0,0 +1,274 @@
...
+    'headers' => ['Authorization' => 'Bearer ' . $appToken],
+]);
+```
+
+### Use case 1: Collect product variation information - all levels
+
+#### 1. Collect product models
+##### 1.1 You are following the App workflow?
+
+In the guided tutorial How to get families and attributes, we have stored a family_code_list. It’s time to use it!
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:122]: missing link

Comment (tomglvng) [reply content/tutorials/guides/how-to-collect-product-variations.md:122]: Ok, it was not precise in the Notion. Sorry!

File: content/tutorials/guides/how-to-collect-product-variations.md (hunk 11 above)
Comment (tomglvng) [reply content/tutorials/guides/how-to-collect-product-variations.md:202]: Ok, it was not precise in the Notion. Sorry!

File: content/tutorials/guides/how-to-collect-product-variations.md (hunk 4 above)
Comment (samokiss) [reply content/tutorials/guides/how-to-collect-product-variations.md:156]: forgetting

File: content/tutorials/guides/how-to-collect-product-variations.md (hunk 5 above)
Comment (samokiss) [reply content/tutorials/guides/how-to-collect-product-variations.md:195]: forget to delete from original snippet

File: content/tutorials/guides/how-to-collect-product-variations.md (hunk 11 above)
Comment (samokiss) [reply content/tutorials/guides/how-to-collect-product-variations.md:202]: done

File: content/tutorials/guides/how-to-collect-product-variations.md (hunk 12 above)
Comment (samokiss) [reply content/tutorials/guides/how-to-collect-product-variations.md:122]: done

File: content/tutorials/guides/how-to-collect-product-variations.md (hunk 10 above)
Comment (samokiss) [reply content/tutorials/guides/how-to-collect-product-variations.md:207]: done

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 13]
```diff
@@ -0,0 +1,277 @@
...
+            </div>
+            <div class="tag-not-selectable">
+                <div class="tag-color tag-color-light-green"></div>
+                <div class="tag-label">Variant Products</div>
+            </div>
+        </td>
+    </tr>
+</table>
+
+<div class="endpoint-container">
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:35]: API endpoints aren't correct, see the Notion page.
https://api.akeneo.com/api-reference.html#get_products_uuid

https://api.akeneo.com/api-reference.html#get_product_models

https://api.akeneo.com/api-reference.html#get_families__family_code__variants__code__

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 14]
```diff
@@ -0,0 +1,277 @@
...
+                <li>Step 4. <a href="how-to-get-product-information.html" target="_blank" rel="noopener noreferrer">How to get product information</a></li>
+            </ul>
+        </div>
+    </div>
+</div>
+
+## Context
+
+In the PIM we handle product models and product variations.
+
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:62]: Miss the tip "Before digging into the code you can find out more about these concepts in our [helpcenter](https://help.akeneo.com/pim/serenity/articles/what-about-products-variants.html#about-products-with-variants)." (See Notion)

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 15]
```diff
@@ -0,0 +1,277 @@
...
+- **Use case 1**: Collect all product variation information, just as they are in the PIM (up to 2 possible levels)
+- **Use case 2**: Collect all product variation information on 1 level only)
+
+::: info
+We noticed that many e-commerce solutions understand product variation on only one level. This means that for Akeneo a special recollection of the variations must be done to have them all on the same level.
+:::
+
+### 0 - Initialization
+
+```php [activate:PHP]
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:102]: This step in not exactly the same than in https://github.com/akeneo/php-guided-tutorials/pull/4/files#diff-4a45687bf67299eda240500899b81b7f57dae3b1014be4a7fe2f7acb7fbd8ac5R11 or https://github.com/akeneo/php-guided-tutorials/pull/4/files#diff-6637c0fc4b1cf3c5db58fbc4fda9e4fc0898392f8bb76e81597e14b08daea23cR29 . I let you harmonize it

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 16]
```diff
@@ -0,0 +1,277 @@
...
+}
+
+```
+
+### Use case 1: Collect product variation information - all levels
+
+#### 1. Collect product models
+##### 1.1 You are following the App workflow?
+
+In the guided tutorial [**"How to get families and attributes"**](/tutorials/how-to-get-families-and-attributes.html), we have stored a family_code_list. It’s time to use it!
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:129]:
```suggestion
In the guided tutorial [**"How to get families and attributes"**](/tutorials/how-to-get-families-and-attributes.html), we have stored a **family_code_list**. It’s time to use it!
```

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 17]
```diff
@@ -0,0 +1,277 @@
...
+{
+    $client = getApiClient();
+
+    $maxProductsPerPage = 100;
+    $maxFamiliesPerQuery = 3;
+    $scope = 'ecommerce';
+
+    // Get family $codes and locales from storage
+    $familyCodes = getFamilyCodes();
+    $locales = getLocales('fr');
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:143]: You should get your locale from your storage

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 18]
```diff
@@ -0,0 +1,277 @@
...
+    saveProductModels($productModels);
+}
+
+```
+
+#### 2. Process product model
+##### 2.1. Parse and store the product model
+Parse and store a product or a product model is definitely the same thing. Please have a how to our guided tutorial [**"How to get product information"**](/tutorials/how-to-collect-products.html).
+
+##### 2.2. Parse and store the product model
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:208]: Not the good title

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 19]
```diff
@@ -0,0 +1,277 @@
...
+
+```
+
+##### 2.2. Collect its product variants
+
+To get product variants associated to a product model, ask to the API
+
+```php [activate:PHP]
+
+function getProductsUuid(): array
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:250]:
```suggestion
function getProductVariants(): array
```

**Review by tomglvng — CHANGES_REQUESTED** [review]

File: content/tutorials/guides/how-to-collect-product-variations.md (hunk 17 above)
Comment (samokiss) [reply content/tutorials/guides/how-to-collect-product-variations.md:143]: that's exactly what I do

File: content/tutorials/guides/how-to-collect-product-variations.md (hunk 13 above)
Comment (samokiss) [reply content/tutorials/guides/how-to-collect-product-variations.md:35]: products
product models
family variants

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 20]
```diff
@@ -0,0 +1,285 @@
...
+</div>
+
+## Context
+
+In the PIM we handle product models and product variations.
+
+![scheme_variants](../../img/tutorials/how-to-collect-product-variations/scheme_variants.png)
+
+::: tips
+Before digging into the code you can find out more about these concepts in our [helpcenter](https://help.akeneo.com/pim/serenity/articles/what-about-products-variants.html#about-products-with-variants).
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:66]:
```suggestion
Before digging into the code you can find out more about these concepts in our <a href="https://help.akeneo.com/pim/serenity/articles/what-about-products-variants.html#about-products-with-variants" target="_blank" rel="noopener noreferrer">helpcenter</a>.
```

Should open a new tab

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 21]
```diff
@@ -0,0 +1,285 @@
...
+$appToken = 'your_app_token'; // Token provided during oAuth steps
+
+// If you haven't done it yet, please follow the Guzzle official documentation for installing the client 
+// https://docs.guzzlephp.org/en/stable/overview.html#installation
+
+// Set your client for querying Akeneo API as follows
+function buildApiClient(): GuzzleHttp\Client
+{
+    $pimUrl = '<PIM_URL>';
+    $appToken = '<APP_TOKEN>'; // Token provided during oauth steps
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:117]:
```suggestion
   $pimUrl = 'https://url-of-your-pim.com';
   $appToken = 'your_app_token'; // Token provided during oAuth steps
```

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 22]
```diff
@@ -0,0 +1,285 @@
...
+
+```php [activate:PHP]
+
+$pimUrl = 'https://url-of-your-pim.com';
+$appToken = 'your_app_token'; // Token provided during oAuth steps
+
+// If you haven't done it yet, please follow the Guzzle official documentation for installing the client 
+// https://docs.guzzlephp.org/en/stable/overview.html#installation
+
+// Set your client for querying Akeneo API as follows
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:113]:
```suggestion
```

You don't need anymore

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 23]
```diff
@@ -0,0 +1,285 @@
...
+{
+    $client = buildApiClient();
+
+    $maxProductsPerPage = 100;
+    $maxFamiliesPerQuery = 3;
+    $scope = 'ecommerce';
+
+    // Get family $codes and locales from storage
+    $familyCodes = getFamilyCodes();
+    $locales = getLocales('fr');
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:149]:
```suggestion
   // Get locales from the storage
    $locales = getLocales();
```

**Review by tomglvng — CHANGES_REQUESTED** [review]

**Pull request closed without merging**